Sources/Sentry/SentryExtraContextProvider.h

```diff
@@ -1,18 +1,31 @@
-#import <Foundation/Foundation.h>
+#if __has_include(<Sentry/SentryDefines.h>)
+#    import <Sentry/SentryDefines.h>
+#else
+#    import "SentryDefines.h"
+#endif
 
 @class SentryCrashWrapper;
 @class SentryNSProcessInfoWrapper;
 @class SentryUIDeviceWrapper;
 
+#if TARGET_OS_IOS && SENTRY_HAS_UIKIT
+@class SentryUIDeviceWrapper;
+#endif // TARGET_OS_IOS && SENTRY_HAS_UIKIT
+
 NS_ASSUME_NONNULL_BEGIN
 
 /**
  * Provider of dynamic context data that we need to read at the time of an exception.
  */
 @interface SentryExtraContextProvider : NSObject
+SENTRY_NO_INIT
 
 - (instancetype)initWithCrashWrapper:(SentryCrashWrapper *)crashWrapper
-                  processInfoWrapper:(SentryNSProcessInfoWrapper *)processInfoWrapper;
+                  processInfoWrapper:(SentryNSProcessInfoWrapper *)processInfoWrapper
+#if TARGET_OS_IOS && SENTRY_HAS_UIKIT
```

Comment:
`l`: These mixed constructors could cause annoying compiler issues. Maybe two separate initializers are easier to maintain. No strong opinion here, I'll leave it to you.

Reply:
Which annoying compiler issues have you encountered in the past? I would keep it for now.